### PR TITLE
Configuration typo fixes

### PR DESCRIPTION
**Pull request by Roiocam**
Separate PR and cherry-pick from #1096, and solve all module typos.

### CODE REVIEW

File: cluster-sharding/src/main/resources/reference.conf [hunk 1]
```diff
@@ -22,7 +22,7 @@ pekko.cluster.sharding {
 
   # When this is set to 'on' the active entity actors will automatically be restarted
   # upon Shard restart. i.e. if the Shard is started on a different ShardRegion
-  # due to rebalance or crash.
+  # due to rebalanced or crash.
```
Comment (pjfanning) [inline cluster-sharding/src/main/resources/reference.conf:25]: better to say `due to a rebalance event or a crash`

File: cluster/src/main/resources/reference.conf [hunk 2]
```diff
@@ -233,7 +233,7 @@ pekko {
     multi-data-center {
 
       # Defines which data center this node belongs to. It is typically used to make islands of the
-      # cluster that are colocated. This can be used to make the cluster aware that it is running
+      # cluster that are collocated. This can be used to make the cluster aware that it is running
```
Comment (pjfanning) [inline cluster/src/main/resources/reference.conf:236]: the change is not wrong - should be colocated or co-located

File: remote/src/main/resources/reference.conf [hunk 3]
```diff
@@ -492,7 +492,7 @@ pekko {
         applied-adapters = []
 
         # The default remote server port clients should connect to.
-        # Default is 7355 (PEKK on a telephone keypad), use 0 if you want a random available port
```
Comment (pjfanning) [inline remote/src/main/resources/reference.conf:495]: 7355 is PEKK - 4 digits, 4 letters - so can you undo this change?

File: remote/src/main/resources/reference.conf [hunk 4]
```diff
@@ -852,7 +852,7 @@ pekko {
         # collected, which is not as efficient as reusing buffers in the pool.
         large-buffer-pool-size = 32
 
-        # For enabling testing features, such as blackhole in pekko-remote-testkit.
+        # For enabling testing features, such as black-hole in pekko-remote-testkit.
```
Comment (pjfanning) [inline remote/src/main/resources/reference.conf:855]: blackhole is correct, don't change this

File: remote/src/main/resources/reference.conf (hunk 3 above)
Comment (Roiocam) [reply remote/src/main/resources/reference.conf]: Thanks, I forgot the previous discussion of pekko default listener port.

File: remote/src/main/resources/reference.conf (hunk 4 above)
Comment (Roiocam) [reply remote/src/main/resources/reference.conf]: Yes, but I think the `black-hole` may clear that `blackhole`, Never mind, let we keep `blackhole`.

**Review by pjfanning — APPROVED** [review]
lgtm
